### PR TITLE
feat(metricbeat): annotation support for daemonset and deployment

### PR DESCRIPTION
**Pull request by kernkonzentrat**
- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

Add annotation support for daemonset and deployment. Annotations are only added to the k8s objects, if values are set for annotations. PR for issue #684

### CODE REVIEW

**Comment by fatmcgav** [comment]
Jenkins test this please

File: metricbeat/tests/metricbeat_test.py [hunk 1]
```diff
@@ -1123,3 +1123,25 @@ def test_setting_fullnameOverride():
...
+    assert r["daemonset"][name]["metadata"]["annotations"]["foo"] == "bar"
+    assert "annotations" not in r["deployment"][name + "-metrics"]["metadata"]
+    config = """
+deployment:
+    annotations:
+        grault: "waldo"
+"""
+    r = helm_template(config)
+    assert "grault" in r["deployment"][name + "-metrics"]["metadata"]["annotations"]
+    assert r["deployment"][name + "-metrics"]["metadata"]["annotations"]["grault"] == "waldo"
```
Comment (fatmcgav) [inline metricbeat/tests/metricbeat_test.py:1145]: Looks like there's one Python lint issue here:

```suggestion
    assert (
        r["deployment"][name + "-metrics"]["metadata"]["annotations"]["grault"]
        == "waldo"
    )
```

Comment (kernkonzentrat) [reply metricbeat/tests/metricbeat_test.py]: @fatmcgav ...added the suggested change to `daemonset` and `deployment` test

**Comment by fatmcgav** [comment]
Jenkins test this please

File: metricbeat/tests/metricbeat_test.py (hunk 1 above)
Comment (fatmcgav) [reply metricbeat/tests/metricbeat_test.py]: @kernkonzentrat It looks like the lint step is still not liking this format:

https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+lint-python/346/label=docker&&virtual&&ubuntu-18.04/console

Comment (kernkonzentrat) [reply metricbeat/tests/metricbeat_test.py]: I formatted the tests with `black` now. I guess it will work eventually.

**Comment by kernkonzentrat** [comment]
Jenkins test this please

**Comment by fatmcgav** [comment]
@kernkonzentrat Thanks for that.

Jenkins test this please

**Comment by kernkonzentrat** [comment]
@fatmcgav `black` still wants to reformat...I can reproduce this locally, trying to fix it and let you know once it's done

**Comment by fatmcgav** [comment]
Jenkins test this please

**Comment by fatmcgav** [comment]
Thanks for persevering there @kernkonzentrat, CI passed on that run :)

**Comment by fatmcgav** [comment]
@kernkonzentrat Thanks again for the contribution.

I've back-ported the changes onto the `7.8` and `6.8` branches, so these changes will be included in the next `7.8` patch release.

**Comment by kernkonzentrat** [comment]
@fatmcgav Thank you for support and reviewing...the project team is looking forward to getting in the update :)

**Pull request merged**